spec.md

```diff
@@ -2041,6 +2100,72 @@ Condition | gRPC Code | Description | Recovery Behavior
+* `NodeExpandVolume` however MAY be called before or after `NodePublishVolume`.
+
+`NodeExpandVolume` expects volume to be available on specified `volume_path` and `NodeExpandVolume` itself will not implicitly call `NodeStageVolume` or `NodePublishVolume`.
+
+`NodeExpandVolume` may also be called to perform online expansion of in-use Volume.
+A plugin may return `FAILED_PRECONDITION` error if underlying volume can not be expanded when in-use.
+
+Examples:
+* Offline Volume Expansion:
+  Given a ElasticSearch process that runs on EBS volume and needs more space.
+  - The administrator takes the elasticsearch server offline and request more space for volume from the CO.
+  - The CO in turn first makes `ControllerExpandVolume` RPC call which results in requesting more space from AWS-EBS for volume-id that was being used by ElasticSearch.
+  - Once `ControllerExpandVolume` finishes the CO will inform administrator about it and administrator will resume the ElasticSearch workload.
+  - On node where ElasticSearch workload is scheduled when workload starts after `NodeStageVolume` the CO calls `NodeExpandVolume`.
+  - Calling `NodeExpandVolume` on volume results in expanding the underlying file system and added space becomes available to workload when it startsup.
+* Online Volume Expansion:
+  Given a Mysql server running using Openstack Cinder and needs more space.
+  - The administrator requests more space for volume from the CO.
+  - The CO in turn first makes `ControllerExpandVolume` RPC call which results in requesting more space from Openstack Cinder for given volume.
+  - The node where mysql workload is running the CO makes `NodeExpandVolume` RPC call while volume is in-use using path where volume is staged.
+  - Calling `NodeExpandVolume` on volume results in expanding the underlying file system and added space automatically becomes available to mysql workload without any downtime.
+
+```protobuf
+message NodeExpandVolumeRequest {
+  // The ID of the volume. This field is REQUIRED.
+  string volume_id = 1;
+
+  // The path on which volume is available.
+  // This field is REQUIRED.
+  string volume_path = 2;
+
+  // This field is OPTIONAL. This allows CO to specify the
+  // capacity requirements of the volume after expansion.
+  CapacityRange capacity_range = 3;
+}
+
+message NodeExpandVolumeResponse {
+  // The capacity of the volume in bytes.
+  // This field is OPTIONAL.
+  int64 capacity_bytes = 1;
+}
+```
+
+##### NodeExpandVolume Errors
+
+| Condition             | gRPC code | Description           | Recovery Behavior                 |
+|-----------------------|-----------|-----------------------|-----------------------------------|
+| Volume in use | 9 FAILED_PRECONDITION | Indicates that the volume corresponding to the specified `volume_id` could not be expanded because it is in use. | Caller SHOULD ensure that volume is not in-use and retry with exponential back off. |
```

Comment:
Is this error for scenarios where volume expansion is not supported in "online" mode?